src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1035,6 +1035,15 @@
     ["SELECT timestamp FROM factsets WHERE certname=? ORDER BY timestamp DESC" certname]
     (:timestamp (first result-set))))
 
+
+(pls/defn-validated factset-producer-timestamp :- (s/maybe pls/Timestamp)
+  "Return the factset producer-timestamp for the given certname, nil if not found"
+  [certname :- String]
+  (sql/with-query-results result-set
+    ["SELECT producer_timestamp FROM factsets WHERE certname=? ORDER BY producer_timestamp DESC" certname]
```

Comment:
this should only ever return one element -- I don't think the order by is needed